core/src/main/java/software/amazon/awssdk/core/protocol/json/internal/JsonProtocolMarshaller.java

```diff
@@ -183,7 +184,7 @@ private void marshallBinaryPayload(Object val) {
                 request.addHeader("Content-Length", Integer.toString(content.length));
             }
         }
-        if (!request.getHeaders().containsKey("Content-Type")) {
+        if (!request.getHeaders().containsKey("Content-Type") && !StringUtils.isEmpty(contentType)) {
```

Comment:
Could we use `StringUtils.isNotBlank` here instead? I know it's slightly different than `isEmpty` in that it would match " " as well as "" but I feel like that's probably the 'same' in this case. A `Content-Type` of only whitespace is no more valid than a `null` or empty `Content-Type`.

You could static import too.